contract_manager/scripts/list_wormhole_contracts.ts

```diff
@@ -0,0 +1,40 @@
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import { DefaultStore, EvmWormholeContract } from "../src";
+
+const parser = yargs(hideBin(process.argv))
+  .usage("Usage: $0")
+  .options({
+    testnet: {
+      type: "boolean",
+      default: false,
+      desc: "Fetch testnet contracts instead of mainnet",
+    },
+  });
+
+async function main() {
+  const argv = await parser.argv;
+  const entries = [];
+  for (const contract of Object.values(DefaultStore.wormhole_contracts)) {
+    if (contract instanceof EvmWormholeContract) {
+      if (contract.getChain().isMainnet() === argv.testnet) continue;
+
+      try {
+        const index = await contract.getCurrentGuardianSetIndex();
+        const chainId = await contract.getChainId();
+        entries.push({
+          chain: contract.getChain().getId(),
+          contract: contract.address,
+          guardianSetIndex: index,
+          chainId: chainId,
+        });
+        console.log(`Fetched contract for ${contract.getId()}`);
+      } catch (e) {
+        console.error(`Error fetching contract for ${contract.getId()}`, e);
+      }
+    }
+  }
```

Comment:
You can restructure this to parallelize out all the contracts too by doing something like:

```ts
const contracts = await Promise.all(Object.values(
    DefaultStore.wormhole_contracts)
        .filter(contract => contract instanceof EvmWormholeContract && contract.getChain().isMainnet() !== argv.testnet)
        .map(async (contract) => {
            const [index, chainId] = await Promise.all(...);
            ...
        })
));
```